pennylane/io.py

```diff
@@ -485,12 +496,23 @@ def circuit():
         quantum_circuit (str): a QASM string containing a valid quantum circuit
         measurements (None | MeasurementProcess | list[MeasurementProcess]): an optional PennyLane
             measurement or list of PennyLane measurements that overrides any terminal measurements
-            that may be present in the input circuit
+            that may be present in the input circuit. If set to ``None``, existing measurements
+            in the input circuit will be used.
 
     Returns:
         function: the PennyLane template created based on the QASM string
+
     """
     plugin_converter = plugin_converters["qasm"].load()
+    if measurements is False:
+        warnings.warn(
+            "The current default behaviour of removing measurements in the QASM code is "
+            "deprecated. Set measurements=None to keep the existing measurements in the QASM "
+            "code or set measurements=[] to remove them from the returned circuit. Starting "
+            "in version 0.38, measurements=None will be the new default.",
+            qml.PennyLaneDeprecationWarning,
+        )
+        measurements = []
```

Comment:
Here I was expecting to only raise a warning if `measurements is False` and there are measurements present in the QASM circuit. Is that something we could determine?

Reply:
I'll make another PR to do that.